compiler/rustc_next_trait_solver/src/solve/eval_ctxt/canonical.rs

```diff
@@ -157,6 +157,13 @@ where
             },
         );
 
+        let num_non_region_vars = canonical.variables.iter().filter(|c| !c.is_region()).count();
```

Comment:
comment pls -- perhaps even `// HACK` or something, idk how permanent you expect this to be in the solver.